controllers/csi/provisioner/reconciler.go

```diff
@@ -111,25 +109,32 @@ func (r *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.R
 
 	tenant, err := r.db.GetTenant(dk.ConnectionInfo().TenantUUID)
 	if err != nil {
-		return reconcile.Result{}, err
+		return reconcile.Result{}, errors.WithStack(err)
 	}
+
 	// Incase of a new tenant
 	if tenant == nil {
 		tenant = &metadata.Tenant{TenantUUID: dk.ConnectionInfo().TenantUUID}
 	}
+	rlog.Info("checking tenant", "uuid", tenant.TenantUUID, "version", tenant.LatestVersion)
 	oldTenant := *tenant
 
 	if err = r.createCSIDirectories(r.path.EnvDir(tenant.TenantUUID)); err != nil {
-		return reconcile.Result{}, err
+		rlog.Error(err, "error when creating csi directories", "path", r.path.EnvDir(tenant.TenantUUID))
+		return reconcile.Result{}, errors.WithStack(err)
 	}
 
+	rlog.Info("csi directories exist", "path", r.path.EnvDir(tenant.TenantUUID))
 	tenant.Dynakube = dk.Name
 
-	if err = r.updateAgent(dk, tenant, dtc, rlog); err != nil {
-		return reconcile.Result{}, err
+	installAgentCfg := newInstallAgentConfig(rlog, dtc, r.path, r.fs, r.recorder, tenant, dk)
+	if err = installAgentCfg.updateAgent(); err != nil {
+		rlog.Error(err, "error when updating agent")
```

Comment:
if it is only logged as an info message, the "level" with which it is logged will also be "info", although it should really be an "error" in this case imo.

Reply:
n.b.: after some discussion we decided to log it as an info, since it is technically not an error, as the csi driver is still in a valid state if it cannot update the agent.